src/components/EventInviteDialog/EventInviteDialog.js

```diff
@@ -7,102 +7,116 @@ import Button from 'components/Button'
 import styles from './EventInviteDialog.scss'
 import { humanResponse } from 'store/models/EventInvitation'
 import TextInput from 'components/TextInput'
+import useInView from 'react-cool-inview'
+import Loading from 'components/Loading'
+const pageSize = 30
 
-export default class EventInviteDialog extends React.PureComponent {
-  state = {
-    invitedIds: [],
-    searchTerm: ''
-  }
+export default function EventInviteDialog ({
+  fetchPeople,
+  forGroups,
+  eventInvitations,
+  people,
+  eventId,
+  onClose,
+  invitePeopleToEvent,
+  pending
+}) {
+  const [invitedIds, setInvitedIds] = useState([])
+  const [searchTerm, setSearchTerm] = useState('')
+  const [pageFetched, setPageFetched] = useState(0)
+
+  const toggleInvite = id => (invitedIds.includes(id))
+    ? setInvitedIds(invitedIds.filter(invitedId => invitedId !== id))
+    : setInvitedIds(invitedIds.concat([id]))
 
-  toggleInvite = id => {
-    const { invitedIds } = this.state
+  const onSearchChange = ({ target: { value } }) => setSearchTerm(value)
 
-    if (invitedIds.includes(id)) {
-      this.setState({
-        invitedIds: invitedIds.filter(invitedId => invitedId !== id)
-      })
-    } else {
-      this.setState({
-        invitedIds: invitedIds.concat([id])
-      })
+  useEffect(() => {
+    const fetch = () => {
+      const forGroupIds = forGroups.map(c => c.id)
+      fetchPeople({ autocomplete: searchTerm, groupIds: forGroupIds, first: pageSize, offset: 0 })
+      setPageFetched(pageSize)
     }
-  }
+    fetch()
+  }, [searchTerm])
 
-  onSearchChange = ({ target: { value } }) => {
-    const { fetchPeople, forGroups } = this.props
-    const forGroupIds = forGroups.map(c => c.id)
-    this.setState({ searchTerm: value })
-    fetchPeople(value, forGroupIds)
-  }
+  const { observe } = useInView({
+    onEnter: () => {
+      const fetch = () => {
+        const forGroupIds = forGroups.map(c => c.id)
```

Comment:
isnt this basically a duplicate function of the one above? could pull out and reuse

Reply:
the two `fetch` functions overall or the `forGroupIds`